### ISSUE TITLE
`minishift delete` fails when a snapshot was made using virsh (kvm/libvirt)

### ISSUE THREAD

**Issue by gbraad** [minishift/minishift#386]
Developers try everything... and they might create snapshots for some reason. At least I did for debugging our own issue

## Scenario

```gherkin
Given a running minishift instance
  And I created some snapshosts of the running instance using `sudo virsh snapshot-create-as minishift snapshot1 "before deploy" --disk-only --atomic`
 When I want to undefine the instance using `minishift delete`
 Then the minishift instance should be available with `sudo virsh list`
```

## Actual result
```
$ minishift delete
Deleting the Minishift VM...
Error deleting the VM:  [Code-55] [Domain-10] Requested operation is not valid: cannot delete inactive domain with 4 snapshots
```

## Explanation
The minishift instance has the disk image and snapshots stored in `~/.minishift/machines`

Doing the following reveals the problem
```
$ sudo virsh snapshot-delete minishift snapshot3`
error: Failed to delete snapshot snapshot3
error: unsupported configuration: deletion of 1 external disk snapshots not supported yet
```

## Solution
The snaphosts are stored alongside of the vitual machine's disk image in `~/.minishift/machines`, but the definitions live in `/var/lib/libvirt/qemu/snapshot/minishift`. They can be removed using:

```
$ virsh snapshot-delete --metadata
```

after which we manually need to remove the snapshots, or:
```
$ rm -rf ~/.minishift/machines
```

**Comment by gbraad** [comment]
-> to be included in https://github.com/minishift/minishift/blob/master/docs/troubleshooting.md

**Comment by hferentschik** [comment]
Gee, well done ;-)

**Comment by hferentschik** [comment]
Not sure whether I consider this a bug per se though?

**Comment by gbraad** [comment]
Not a bug, rather a usage we have not foreseen:

We can instruct people to not use metadata it seems:
```
$ virsh snapshot-create-as --domain vm1 overlay1 --diskspec vda,file=/export/overlay1.qcow2 --disk-only --atomic --no-metadata
```

**Comment by thatdocslady** [comment]
If I understand correctly this looks like something that should be documented in the troubleshooting guide, as it describes a less-than-common use-case but not something entirely random that we don't expect anyone to do.

**Comment by agajdosi** [comment]
@gbraad Testing proposed workaround I have found one difference from expected behaviour, see below:

```
[agajdosi@dhcp-24-156 minishift]$ sudo virsh snapshot-delete --metadata minishift snapshot1
Domain snapshot snapshot1 deleted
[agajdosi@dhcp-24-156 minishift]$ rm -rf ~/.minishift/machines
[agajdosi@dhcp-24-156 minishift]$ minishift start
Starting local OpenShift cluster using 'kvm' hypervisor...
E0306 13:14:21.810171   11786 start.go:150] Error starting the VM: Error creating the VM. Error creating machine: Error in driver during machine creation: [Code-9] [Domain-20] operation failed: domain 'minishift' already exists with uuid b9f9ceb0-b6a6-41c1-912c-d5c56a2a6d88. Retrying.
E0306 13:14:22.475735   11786 start.go:150] Error starting the VM: Error starting stopped host: [Code-38] [Domain-18] Cannot access storage file '/home/agajdosi/.minishift/machines/minishift/minishift.snapshot1' (as uid:107, gid:107): No such file or directory. Retrying.
E0306 13:14:22.521561   11786 start.go:150] Error starting the VM: Error starting stopped host: [Code-38] [Domain-18] Cannot access storage file '/home/agajdosi/.minishift/machines/minishift/minishift.snapshot1' (as uid:107, gid:107): No such file or directory. Retrying.
E0306 13:14:22.521591   11786 start.go:156] Error starting the VM:  Error creating the VM. Error creating machine: Error in driver during machine creation: [Code-9] [Domain-20] operation failed: domain 'minishift' already exists with uuid b9f9ceb0-b6a6-41c1-912c-d5c56a2a6d88
Error starting stopped host: [Code-38] [Domain-18] Cannot access storage file '/home/agajdosi/.minishift/machines/minishift/minishift.snapshot1' (as uid:107, gid:107): No such file or directory
Error starting stopped host: [Code-38] [Domain-18] Cannot access storage file '/home/agajdosi/.minishift/machines/minishift/minishift.snapshot1' (as uid:107, gid:107): No such file or directory
[agajdosi@dhcp-24-156 minishift]$ minishift delete
Deleting the Minishift VM...
Minishift VM deleted.
[agajdosi@dhcp-24-156 minishift]$ minishift start
Starting local OpenShift cluster using 'kvm' hypervisor...
Provisioning OpenShift via [...]
```

**Comment by gbraad** [comment]
@agajdosi this can be covered with
```
$ sudo virsh undefine minishift
```

but maybe do not do the `rm -rf`, but instead do `minishift delete` after removing all the snapshots.

**Comment by agajdosi** [comment]
@gbraad Thank you for information. It works for me. @Preeticp, for me complete procedure working is then this:

```
[agajdosi@dhcp-24-156 minishift]$ sudo virsh snapshot-delete --metadata minishift snapshot1
Domain snapshot snapshot1 deleted
[agajdosi@dhcp-24-156 minishift]$ sudo virsh undefine minishift
Domain minishift has been undefined
[agajdosi@dhcp-24-156 minishift]$ minishift start
Starting local OpenShift cluster using 'kvm' hypervisor...
Provisioning OpenShift via [...]
```

**Comment by Preeticp** [comment]
Resolved with #528
